internal/registration/registration.go

```diff
@@ -82,30 +121,77 @@ func (r *Registration) registerDeviceWithRetries(interval int64) {
 }
 
 func (r *Registration) registerDeviceOnce() error {
+
+	key, csr, err := r.renewCertificate()
+	if err != nil {
+		return err
+	}
+
 	hardwareInformation, err := r.hardware.GetHardwareInformation()
 	if err != nil {
 		return err
 	}
+
 	registrationInfo := models.RegistrationInfo{
-		Hardware: hardwareInformation,
+		Hardware:           hardwareInformation,
+		CertificateRequest: string(csr),
 	}
+
 	content, err := json.Marshal(registrationInfo)
 	if err != nil {
 		return err
 	}
+
 	data := &pb.Data{
 		MessageId: uuid.New().String(),
 		Content:   content,
 		Directive: "registration",
 	}
-
 	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
 	defer cancel()
 
 	// Call "Send"
-	if _, err := r.dispatcherClient.Send(ctx, data); err != nil {
+	response, err := r.dispatcherClient.Send(ctx, data)
+	if err != nil {
+		return err
+	}
+	var parsedResponse YGGDresponse
+	err = json.Unmarshal(response.Response, &parsedResponse)
+	if err != nil {
+		return err
+	}
+
+	if parsedResponse.StatusCode >= 300 {
+		return fmt.Errorf("cannot register to the operator")
+	}
+
+	var message models.MessageResponse
+	err = json.Unmarshal(parsedResponse.Body, &message)
+	if err != nil {
+		return err
+	}
+
+	parsedContent, ok := message.Content.(map[string]interface{})
+	if !ok {
+		return fmt.Errorf("Cannot parse message content, content=%+v", message.Content)
+	}
+
+	cert, ok := parsedContent["certificate"]
+	if !ok {
+		return fmt.Errorf("Cannot retrieve certificate from parsedResponse, content=%+v", message.Content)
+	}
```

Comment:
Do we wish to print the entire error content to log?couldn't it contain sensitive data?